extern/CMakeProject-mad.cmake

```diff
@@ -1,52 +1,88 @@
-list(APPEND MAD_SRC
-  "mad-0.15.1b/bit.c"
-  "mad-0.15.1b/decoder.c"
-  "mad-0.15.1b/fixed.c"
-  "mad-0.15.1b/frame.c"
-  "mad-0.15.1b/huffman.c"
-  "mad-0.15.1b/layer12.c"
-  "mad-0.15.1b/layer3.c"
-  "mad-0.15.1b/stream.c"
-  "mad-0.15.1b/synth.c"
-  "mad-0.15.1b/timer.c"
-  "mad-0.15.1b/version.c"
+set(MAD_DIR "${SM_EXTERN_DIR}/libmad-0.15.1b")
+
+check_include_file(assert.h HAVE_ASSERT_H)
+check_include_file(dlfcn.h HAVE_DLFCN_H)
```

Comment:
Some of the checks here are redundant, and would not be checked a second time. This is not the correct spot for placing these.